scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -886,6 +886,11 @@ def apply_filters(self, **kwargs):
                     if dep_intersection != set(ts.depends_on):
                         instance.add_filter("No hardware support", Filters.PLATFORM)
 
+                if ts.depends_on_anyof:
+                    dep_intersection = ts.depends_on_anyof.intersection(set(plat.supported))
+                    if not dep_intersection:
+                        instance.add_filter("No hardware support", Filters.PLATFORM)
```

Comment:
a bit different description from what 'depends_on' has, like
```suggestion
                        instance.add_filter("No hardware support (depends_on_anyof)", Filters.PLATFORM)
```